Fix: detect all font types on Windows

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Fix - enumerate all font styles to detect font variants on all Windows versions to detect all fonts.
#### Motivation for adding to Mudlet
Fixes #8248
#### Other info (issues closed, discussion etc)

claude committed:
Fix: Enumerate all font styles to detect font variants on Windows

Fixes #8248

Issue:
On Windows 10 64-bit and Windows 11, getAvailableFonts() was not
detecting all font variants. For example, when "EB Garamond SemiBold"
was installed, only "EB Garamond" appeared in the available fonts list.
This occurred because QFontDatabase::families() only returns unique
family names, and when multiple font weights share the same family name,
only one entry is returned.

Solution:
1. Enhanced getAvailableFonts() to enumerate not just font families,
   but also all available styles for each family (Bold, SemiBold,
   Italic, etc.)

2. Added mudlet::createFont() helper function that intelligently parses
   font names containing style information (e.g., "EB Garamond SemiBold")
   and uses QFont::setStyleName() to explicitly set the font style

3. Updated setFont() in TLuaInterpreterUI.cpp to use the new helper
   function, ensuring fonts with style names are properly resolved

The fix maintains backward compatibility - existing font names without
styles continue to work, and fonts that previously worked still work.
Users can now access all installed font variants through both
getAvailableFonts() and setFont().

Technical details:
- getAvailableFonts() now returns both base families and "Family Style"
  combinations (skipping "Normal" and "Regular" styles as redundant)
- createFont() performs case-insensitive matching and tries exact family
  match first, then parses "Family Style" format if needed
- Results are sorted case-insensitively for consistency

SlySven commented on src/mudlet.cpp:
```diff
@@ -5554,7 +5556,90 @@ void mudlet::slot_newDataOnHost(const QString& hostName, const bool isLowerPrior
+
+    // If not found, try to parse as "Family Style"
+    // Split on last space to separate family from style
+    const int lastSpaceIndex = fontName.lastIndexOf(QLatin1Char(' '));
```
```suggestion
    const int lastSpaceIndex = fontName.lastIndexOf(QChar::Space);
```

SlySven reviewed (commented):
:thinking: How will this look in locales that do not use spaces between words?

vadi2 committed:
Update src/mudlet.cpp

Co-authored-by: Stephen Lyons <slysven@virginmedia.com>

claude committed:
Fix: Geyser Label font handling to support font style variants

Fixes Geyser's setFont() method to properly handle font names with
style information (e.g., "EB Garamond SemiBold").

Issue:
While getAvailableFonts() now correctly returns font variants like
"EB Garamond SemiBold", Geyser's setFont() method was still using
HTML's <font face=""> attribute, which only accepts base family names.
This caused "EB Garamond SemiBold" to fail while "EB Garamond" worked.

Solution:
1. Added parseFontName() helper function that:
   - Parses "Family Style" format font names
   - Extracts the base family name
   - Maps style names (SemiBold, Bold, Light, Italic, etc.) to CSS
     font-weight and font-style properties

2. Updated setFont() to parse font names and store:
   - fontFamily: base family name
   - fontCssStyle: CSS properties for the style
   - fontHtmlFace: family name for HTML (backward compat)

3. Updated all echo methods (echo, decho, hecho, cecho) to:
   - Use CSS font-family instead of HTML font face attribute
   - Apply font-weight and font-style CSS properties
   - Maintain backward compatibility with base family names

Font weight mapping:
- Thin -> 100, Light -> 300, Medium -> 500
- SemiBold/DemiBold -> 600, Bold -> 700, ExtraBold -> 800
- Black/Heavy -> 900

Font style mapping:
- Italic/Oblique -> font-style: italic

This ensures both "EB Garamond" and "EB Garamond SemiBold" work
correctly in Geyser labels.

vadi2 commented:
No longer needed with variable width fonts.
